### PR TITLE
feat: add native token foreign asset type

### PR DESCRIPTION
**Pull request by wangminqi**
For adding gensis build of asset manager and runtime upgrade to correct old code.

### CODE REVIEW

**Comment by wangminqi** [comment]
The migration relocated is temporary. There will be an extra pr for #821 and #801 .

File: pallets/xcm-asset-manager/src/lib.rs [hunk 1]
```diff
@@ -145,6 +147,8 @@ pub mod pallet {
 		type Currency: ReservableCurrency<Self::AccountId>;
 
 		type WeightInfo: WeightInfo;
+
+		type NativeTokenForeignAssetType: Get<ForeignAssetTypeOf<Self>>;
```
Comment (Kailai-Wang) [inline pallets/xcm-asset-manager/src/lib.rs:149]: It seems this type is never used in the business logic except in the `genesis build`?

File: runtime/litmus/src/lib.rs [hunk 2]
```diff
@@ -125,6 +126,7 @@ pub type Executive = frame_executive::Executive<
 	// it was reverse order before.
 	// See the comment before collation related pallets too.
 	AllPalletsWithSystem,
+	migration::AddNativeTokenForeignAssetTypeForAssetManager<Runtime>,
```
Comment (Kailai-Wang) [inline runtime/litmus/src/lib.rs:135]: I don't think we even need the migration, as no storage has ever been set on-chain in the asset manager. Please let me know if I misunderstand it.

File: pallets/xcm-asset-manager/src/lib.rs (hunk 1 above)
Comment (wangminqi) [reply pallets/xcm-asset-manager/src/lib.rs:149]: yes. For genesis build only.

File: runtime/litmus/src/lib.rs (hunk 2 above)
Comment (wangminqi) [reply runtime/litmus/src/lib.rs:135]: The problem is we miss the genesis build. We need to fix it somehow.

Comment (Kailai-Wang) [reply runtime/litmus/src/lib.rs:135]: But we missed the train of genesis build already 😂  the migration won't help the genesis build -- if you need a different state, we can directly change the pallet setting and/or send on-chain extrinsics, as the current state for it is pretty much empty anyway

File: pallets/xcm-asset-manager/src/lib.rs (hunk 1 above)
Comment (Kailai-Wang) [reply pallets/xcm-asset-manager/src/lib.rs:149]: That doesn't make much sense to me 😂  I would expect something like "you can't register asset type that is equal to NativeTokenForeignAssetType" at least..

Comment (wangminqi) [reply pallets/xcm-asset-manager/src/lib.rs:149]: What you describe is partially true, we actually can not serialize multilocation, so genesis build can not use NativeTokenForeignAssetType directly.

Comment (Kailai-Wang) [reply pallets/xcm-asset-manager/src/lib.rs:149]: I mean if it's only used in genesis build, what's the point of having it in the future logic?

File: runtime/litmus/src/lib.rs (hunk 2 above)
Comment (wangminqi) [reply runtime/litmus/src/lib.rs:135]: Correct!

**Comment by wangminqi** [comment]
This PR is closed. A new approach will be accepted.

**Pull request closed without merging**